### PR TITLE
=pro check binary compatibility against all bc-compatible versions

### PR DESCRIPTION
**Pull request by 2m**
Also verified that BC checks pass with 2.11.

Released a temporary version of mima sbt plugin to a separate bintray repo until official updated version is released.

### CODE REVIEW

**Comment by patriknw** [comment]
LGTM, excellent

**Comment by 2m** [comment]
Failure was due to sbt version update. Same problem as fixed by the #16201. Instead of using the same fix, here we only upgrade to sbt 0.13.5 because we are still not using autoplugins.

File: project/AkkaBuild.scala [hunk 1]
```diff
@@ -1105,33 +1105,49 @@ object AkkaBuild extends Build {
...
       val fullId = crossVersion.value match {
          case _ : CrossVersion.Binary => id + "_" + scalaBinaryVersion.value
          case _ : CrossVersion.Full => id + "_" + scalaVersion.value
-         case CrossVersion.Disabled => id 
+         case CrossVersion.Disabled => id
        }
-      Some(organization.value % fullId % version) // the artifact to compare binary compatibility with
+      // check against all binary compatible artifacts
+      val artifacts = for (version <- first to last)
+        yield organization.value % fullId % s"2.3.$version"
```
Comment (rkuhn) [inline project/AkkaBuild.scala]: wouldn’t `%%` work here as well?

File: project/AkkaBuild.scala [hunk 2]
```diff
@@ -1105,33 +1105,49 @@ object AkkaBuild extends Build {
...
   )
 
-  def akkaPreviousArtifact(id: String): Def.Initialize[Option[sbt.ModuleID]] = Def.setting {
+  def akkaPreviousArtifacts(id: String): Def.Initialize[Set[sbt.ModuleID]] = Def.setting {
     if (enableMiMa) {
       // Note: This is a little gross because we don't have a 2.3.0 release on Scala 2.11.x
       // This should be expanded if there are more deviations.
-      val version: String = 
+      val (first, last) = {
+        val lastAkka = 10
```
Comment (rkuhn) [inline project/AkkaBuild.scala]: could this just be extracted from the current version by subtracting one?

Comment (rkuhn) [inline project/AkkaBuild.scala]: Ah, that would require us to use `2.3.12-SNAPSHOT` instead of `2.3-SNAPSHOT`.

**Comment by rkuhn** [comment]
LGTM! We should wrap this up as a first step and refine the filters to apply only to specific version ranges later.

File: project/AkkaBuild.scala (hunk 2 above)
Comment (patriknw) [inline project/AkkaBuild.scala]: it's 11 now!

File: project/plugins.sbt [hunk 3]
```diff
@@ -11,7 +11,9 @@ addSbtPlugin("com.typesafe.sbt" % "sbt-site" % "0.7.1")
 
 addSbtPlugin("com.typesafe.sbt" % "sbt-osgi" % "0.6.0")
 
-addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.1.6")
+addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.1-b71a3e88f0a21d6ce5efe7c0580dba9784295358")
+
```
Comment (patriknw) [inline project/plugins.sbt]: it's published now?

Comment (ktoso) [inline project/plugins.sbt]: good catch, yes it is.
I made a separate bump PR, https://github.com/akka/akka/pull/17605 so if we include the bump here the other one can be closed

**Comment by patriknw** [comment]
LGTM

**Comment by rkuhn** [comment]
So, is this good to merge?

File: project/plugins.sbt (hunk 3 above)
Comment (2m) [inline project/plugins.sbt]: There is a new version of mima plugins, but it does not have multiple artifacts functionality.

**Comment by 2m** [comment]
Not really. It needs a new release of mima plugin. I will keep this for clean up day.

**Comment by rkuhn** [comment]
Okay, sounds good.

**Comment by 2m** [comment]
I will retarget this to master.

**Pull request closed without merging**